client/app/components/queries/AddToDashboardDialog.jsx

```diff
@@ -0,0 +1,110 @@
+import { isString } from 'lodash';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Modal from 'antd/lib/modal';
+import Input from 'antd/lib/input';
+import List from 'antd/lib/list';
+import Icon from 'antd/lib/icon';
+import { wrap as wrapDialog, DialogPropType } from '@/components/DialogWrapper';
+import { QueryTagsControl } from '@/components/tags-control/TagsControl';
+import { Dashboard } from '@/services/dashboard';
+import notification from '@/services/notification';
+import useSearchResults from '@/lib/hooks/useSearchResults';
+
+import './add-to-dashboard-dialog.less';
+
+function AddToDashboardDialog({ dialog, visualization }) {
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const [doSearch, dashboards, isLoading] = useSearchResults((term) => {
+    if (isString(term) && (term.length >= 3)) {
```

Comment:
The loading indicator was not the only thing that caused me a bit of confusion :confused:, do we really need this `term.length >= 3`? (I mean, the results are paged, right?) There is the option of using Antd form to indicate that minimum length, but I don't think it's that worth it. Well, wdyt?

Reply:
I'm not sure why it was implemented this way; of course that API can search by a single character. Also, here we don't show all results, but only a most relevant ones (20 or 25 - don't remember exactly) - I think it makes sense.

Reply:
I think the idea was to reduce number of requests we send because it's less likely that a <3 characters term will be meaningful. We can drop this.

Reply:
@arikfr @gabrieldutra https://github.com/getredash/redash/pull/4408/commits/40d5aa91f5910de7117842a52406df98e480d52b